pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -223,3 +241,84 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 		By("Test successful !")
 	})
 }
+
+func GetNextAvailableTkrVersion(tkgctlClient tkgctl.TKGClient, currentTkrVersion string) (string, error) {
+	tkrs, err := tkgctlClient.GetTanzuKubernetesReleases("")
+	if err != nil {
+		return "", err
+	}
+
+	foundTkrVersion := ""
+	availableVersions := make(map[string]string)
+	for i := range tkrs {
+		compatible := ""
+		for _, condition := range tkrs[i].Status.Conditions {
+			if condition.Type == runv1alpha3.ConditionCompatible {
+				compatible = string(condition.Status)
+				break
+			}
+		}
+		if !strings.EqualFold(compatible, "true") {
+			continue
+		}
+		labels := tkrs[i].Labels
+		if labels != nil {
```

Comment:
nit: `if tkr.Labels != nil`

The `labels` variable is only used in the block here so its scope should be limited within it

Reply:
good catch. also just like above comment, this way will not copy a variable (allocate more memory)
done